dxp/javascript_minification.md

```diff
@@ -0,0 +1,86 @@
+
+-   [frontend-js-minifier](https://github.com/liferay/liferay-portal/tree/master/modules/apps/frontend-js/frontend-js-minifier): this project contains the two default minifiers that DXP ships out of the box (Google and Yahoo). See the two following sections for more details.
+
+### GoogleJavaScriptMinifier
+
+This is the default DXP minifier (because it is deployed with a higher service ranking than the other ones available).
+
+It uses [Google Closure Compiler](https://github.com/google/closure-compiler) under the hood. Because this is an _internal_ library from Google (thus unsupported and subject to big breaking changes) updating it has usually a significant cost so it is not done very often.
+
+### YahooJavaScriptMinifier
+
+This uses [YUI Compressor](https://github.com/yui/yuicompressor) from Yahoo!, which is still active but unmaintained (except for some PRs that get merged from time to time). The last commit at the time of this writing (Sep 2020) is from May 2019.
+
+Unless someone has a very good reason to use it, this shouldn't be the choice for any production site but it is still shipped for compatibility reasons.
+
+## Components of DXP using the minifier
+
+Currently, the following components in DXP use the minifier services.
+
+### StripFilter
+
+This is a filter used to minify inline JavaScript code contained in `<script>` tags and embedded in the HTML code.
+
+> It also does transformations for `<input>`, `<textarea>`, `<pre>` and `<style>` but that's out of the scope of this document.
+
+It filters any resource with its URL starting with (as specified in [liferay-web.xml](https://github.com/liferay/liferay-portal/blob/master/portal-web/docroot/WEB-INF/liferay-web.xml)):
+
+-   `/c/...`
+-   `/group/...`
+-   `/user/...`
+-   `/web/...`
+
+And it can be configured with:
+
+-   [portal.properties](https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/portal.properties):
+    -   `minifier.inline.content.cache.enabled`: set to `true` to make the filter minify `<script>` tags content (by default `true`).
+    -   `minifier.inline.content.cache.skip.javascript`: this defines a comma separated list of tokens to look for in the `<script>` content so that, if any is found, the minified response is not cached. By default, any `<script>` containing `getSessionId` or `encryptedUserId` is not cached.
+    -   `strip.ignore.paths`: list of paths to ignore (by default `/document_library/get_file`).
+    -   `strip.mime.types` in `portal.properties`: defines the mime types of responses that should be processed by the filter (by default `text/html*` and `text/xml*`). Note that you can put a `*` at the end of the config values to catch all mime types starting with the string (not being exactly the string).
```

Comment:
"mime" → "MIME" in two places

Comma after "default".

> not being exactly the string

This bit isn't clear to me. Does that mean that `*` matches 1 or more characters? (vs. 0 or more). If so, this might be more precise as:

> Note that you can put a `*` at the end of the config values to catch all mime types that have that value _as a prefix_ followed by additional characters (ie. `text/html*` does _not_ match `text/html` alone).